### DISCUSSION TITLE
Using ENV variables instead of mediamtx.yml

### DISCUSSION THREAD

**Discussion by cooperbrown9** [bluenviron/mediamtx#3712] [Questions]
### Question

Hi all, due to some very specific filesystem limitations, I am unable to use the mediamtx.yml for my config. I am trying to translate my mediamtx.yml file to be ENV vars that I pass in through "-e" in my docker run command. 
Here is my mediamtx.yml file (works in dev but not production due to the constraints of the machine I am deploying this one)
```yaml
pathDefaults:
  record: yes
  recordPath: ./recordings/%path/%Y-%m-%d_%H-%M-%S-%f
  recordSegmentDuration: 10s
playback: yes
```

Here is my current docker run command
```sh
docker run --name mediaserver -e MTX_PROTOCOLS=tcp -e hlsAllowOrigin=* -e hlsAllowHeaders=* -e MTX_PATHDEFAULTS_RECORD=yes -e MTX_PATHDEFAULTS_RECORDPATH=recordings/%path/%Y-%m-%d_%H-%M-%S-%f -e MTX_PATHDEFAULTS_RECORDSEGMENTDURATION=10s -e MTX_PLAYBACK=yes -e MTX_PATHS_01_SOURCE=rtsp://${username}:${password}@${LAN}/media/video2 -p 8554:8554 -p 1935:1935 -p 8888:8888 -p 8889:8889 -p 9996:9996 --network=host bluenviron/mediamtx:latest
```

However, I can't even get it to give me segments in 10s increments, so I'm assuming the env var name is wrong. 

To be clear, this works perfectly in DEV when pulling the config from the YML file, but due to the nature of this problem, I cannot use the YML in production and must pass in ENV vars. Does anybody know exactly how to translate that YML to just straight environment variables in the docker run command?

**Comment by aler9** [comment]
Hello, the command you posted seems right to me, in particular, the `MTX_PATHDEFAULTS_RECORDSEGMENTDURATION` is correct.
Keep in mind that recording segments cannot be cut at will, but they can be cut only in the presence of an I-frame. If I-frames are coming at a frequency that is not compatible with your needs, re-encode the stream with FFmpeg or change the "I-frame interval" setting in your camera.
